src/email/email.ts

```diff
@@ -0,0 +1,190 @@
+import { BaseClient } from '../base/client';
+import { SendLayerValidationError } from '../exceptions';
+import * as fs from 'fs';
+import * as path from 'path';
+import axios from 'axios';
+import { URL } from 'url';
+import { createHash } from 'crypto';
+import { EmailRecipient, EmailAttachment, EmailOptions, SendEmailResponse } from '../types';
+
+
+export class NewEmail extends BaseClient {
+  constructor(apiKey: string) {
+    super(apiKey);
+  }
+
+  private validateEmail(email: string): boolean {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return emailRegex.test(email);
+  }
+
+  private validateRecipient(recipient: string | EmailRecipient, recipientType: string = "recipient"): EmailRecipient {
+    if (typeof recipient === 'string') {
+      if (!this.validateEmail(recipient)) {
+        throw new SendLayerValidationError(`Invalid ${recipientType} email address: ${recipient}`);
+      }
+      return { email: recipient };
+    }
+    if (!this.validateEmail(recipient.email)) {
+      throw new SendLayerValidationError(`Invalid ${recipientType} email address: ${recipient.email}`);
+    }
+    return recipient;
+  }
+
+  private validateAttachment(attachment: EmailAttachment): void {
+    if (!attachment.path) {
+      throw new SendLayerValidationError('Attachment path is required');
+    }
+    if (!attachment.type) {
+      throw new SendLayerValidationError('Attachment type is required');
+    }
+  }
+
+  private isUrl(path: string): boolean {
+    try {
+      new URL(path);
+      return true;
+    } catch {
+      return false;
+    }
+  }
+
+  private async readAttachment(filePath: string): Promise<string> {
+    try {
+      // Check if the path is a URL
+      if (this.isUrl(filePath)) {
+        try {
+          const response = await axios.get(filePath, { responseType: 'arraybuffer', timeout: 30000 });
+          const content = response.data;
+          
+          return Buffer.from(content).toString('base64');
+        } catch (error: unknown) {
+          if (error instanceof Error) {
+            throw new SendLayerValidationError(`Error fetching remote file: ${error.message}`);
+          }
+          throw new SendLayerValidationError('Unknown error occurred while fetching remote file');
+        }
+      }
+
+      // Try different local paths
+      const pathsToTry = [
+        filePath, // Original path
+        path.resolve(filePath), // Absolute path
+        path.join(process.cwd(), filePath), // Relative to current working directory
+        path.join(__dirname, filePath), // Relative to script directory
+        path.join(process.env.HOME || '', filePath) // Relative to user's home directory
+      ];
+
+      for (const currentPath of pathsToTry) {
+        try {
+          if (fs.existsSync(currentPath)) {
+            if (!fs.statSync(currentPath).isFile()) {
```

Comment:
This nested `if` looks weird, what happens if `currentPath` doesn't exist? Could the double `if` be merged into a single `&&` condition?